docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md

```diff
@@ -0,0 +1,321 @@
+---
+
+title: In place propagation of changes affecting Kubernetes objects only
+authors:
+- "@fabriziopandini"
+- @sbueringer
+reviewers:
+- @oscar
+- @vincepri
+  creation-date: 2022-02-10
+  last-updated: 2022-02-26
+  status: implementable
+  replaces:
+  superseded-by:
+
+---
+
+# In place propagation of changes affecting Kubernetes objects only
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+**In-place mutable fields**: fields which changes would only impact Kubernetes objects or/and controller behaviour
+but they won't mutate in any way provider infrastructure nor the software running on it. In-place mutable fields
+are propagated in place by CAPI controllers to avoid the more elaborated mechanics of a replace rollout.
+These includes metadata, MinReadySeconds, NodeDrainTimeout, NodeVolumeDetachTimeout and NodeDeletionTimeout but are
+not limited to be expanded in the future.
+
+## Summary
+
+This document discusses how labels, annotation and other fields impacting only Kubernetes objects or controller behaviour (e.g NodeDrainTimeout)
+propagate from ClusterClass to KubeadmControlPlane/MachineDeployments and ultimately to Machines.
+
+## Motivation
+
+Managing labels on Kubernetes nodes has been a long standing [issue](https://github.com/kubernetes-sigs/cluster-api/issues/493) in Cluster API.
+
+The following challenges have been identified through various iterations:
+
+- Define how labels propagate from Machine to Node.
+- Define how labels and annotations propagate from ClusterClass to KubeadmControlPlane/MachineDeployments and ultimately to Machines.
+- Define how to prevent that label and annotation propagation triggers unnecessary rollouts.
+
+The first point is being addressed by [Label Sync Between Machine and underlying Kubernetes Nodes](./20220927-label-sync-between-machine-and-nodes.md),
```

Comment:
This is a broken link, is there an open proposal with this doc?

Reply:
Should be fixed after rebase

Reply:
Works now